include/magic_enum.hpp

```diff
@@ -363,23 +396,24 @@ inline constexpr auto type_name_v = n<E>();
 template <typename E, E V>
 constexpr auto n() noexcept {
   static_assert(is_enum_v<E>, "magic_enum::detail::n requires enum type.");
-  [[maybe_unused]] constexpr auto custom_name = customize::enum_name<E>(V);
+  using std::holds_alternative;
+  using std::get;
 
-  if constexpr (custom_name.empty()) {
-    if constexpr (supported<E>::value) {
+  [[maybe_unused]] constexpr auto custom = customize::enum_name<E>(V);
+  if constexpr (holds_alternative<string_view>(custom)) {
+    constexpr auto name = get<string_view>(custom);
+    return static_string<name.size()>{name};
```

Comment:
I think after this line need a `static_assert` too if the function returns with empty string.
The message is something like: `"Empty string result is not allowed here. Please return with default_tag or invalid_tag"`